tests/route-spec/responses.test.ts

```diff
@@ -0,0 +1,278 @@
+import test from "ava"
+import { createWithEdgeSpec } from "src/create-with-edge-spec"
+import { EdgeSpecRequest, EdgeSpecResponse } from "src/types/web-handler"
+import { z } from "zod"
+
+test("serializes json basic", async (t) => {
+  const withEdgeSpec = createWithEdgeSpec({
+    apiName: "hello-world",
+    productionServerUrl: "https://example.com",
+
+    authMiddlewareMap: {},
+    globalMiddlewares: [],
+  })
+
+  const response = await withEdgeSpec({
+    auth: "none",
+    methods: ["GET"],
+    jsonResponse: z.object({
+      hello: z.string(),
+    }),
+  })(() => {
+    return EdgeSpecResponse.json({
+      hello: "world",
+    })
+  })({} as EdgeSpecRequest)
```

Comment:
I think we have a fixture for this?

Reply:
a fixture for what? I don't want to run a server, I want to test the route function directly

Reply:
unless you think it would be best to run e2e, but then its harder to test errors for instance

Reply:
`tests/fixtures/get-test-route.ts`? idk
still would run in the same VM though so easy to do assertions in route code
just wondering if it's worth avoiding doing `{} as EdgeSpecRequest` in our tests